src/index.js

```diff
@@ -13,7 +13,13 @@ const getDocument = ( () => {
 	let doc;
 	return () => {
 		if ( ! doc ) {
-			doc = document.implementation.createHTMLDocument( '' );
+			if ( process.browser ) {
+				doc = document.implementation.createHTMLDocument( '' );
+			} else {
+				// Use jsdom in Node.js environment to emulate Browser APIs.
+				const DOMImplementation = require( 'jsdom/lib/jsdom/living/generated/DOMImplementation' );
+				doc = DOMImplementation.createImpl().createHTMLDocument( '' );
+			}
```

Comment:
Not sure if you'd consider it any "simpler", but we can consolidate common logic a bit.

```suggestion
			let impl;
			if ( process.browser ) {
				impl = document.implementation;
			} else {
				// Use jsdom in Node.js environment to emulate Browser APIs.
				const DOMImplementation = require( 'jsdom/lib/jsdom/living/generated/DOMImplementation' );
				impl = DOMImplementation.createImpl();
			}
			
			doc = impl.createHTMLDocument( '' );
```

Reply:
Always :)